File: app/src/test/java/com/sakuna63/tumblrin/test/rule/RxRule.java
package com.sakuna63.tumblrin.test.rule;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;

import rx.Scheduler;
import rx.android.plugins.RxAndroidPlugins;
import rx.android.plugins.RxAndroidSchedulersHook;
import rx.android.schedulers.AndroidSchedulers;
import rx.functions.Func1;
import rx.plugins.RxJavaHooks;
import rx.schedulers.Schedulers;

public class RxRule implements TestRule {
    @Override
    public Statement apply(final Statement base, Description description) {
        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                AndroidSchedulers.reset();
                RxAndroidPlugins.getInstance().reset();
                RxAndroidPlugins.getInstance().registerSchedulersHook(new RxAndroidSchedulersHook() {
                    @Override
                    public Scheduler getMainThreadScheduler() {
                        return Schedulers.immediate();
                    }
                });
//
//                RxJavaHooks.reset();
//                RxJavaHooks.setOnIOScheduler(new Func1<Scheduler, Scheduler>() {
//                    @Override
//                    public Scheduler call(Scheduler scheduler) {
//                        return Schedulers.immediate();
//                    }
//                });

                base.evaluate();
            }
        };
    }
}

File: app/src/test/java/com/sakuna63/tumbin/test/rule/RxRule.kt
package com.sakuna63.tumbin.test.rule

import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement
import rx.Scheduler
import rx.android.plugins.RxAndroidPlugins
import rx.android.plugins.RxAndroidSchedulersHook
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class RxRule : TestRule {
    override fun apply(base: Statement, description: Description): Statement {
        return object : Statement() {
            @Throws(Throwable::class)
            override fun evaluate() {
                AndroidSchedulers.reset()
                RxAndroidPlugins.getInstance().reset()
                RxAndroidPlugins.getInstance().registerSchedulersHook(object : RxAndroidSchedulersHook() {
                    override fun getMainThreadScheduler(): Scheduler {
                        return Schedulers.immediate()
                    }
                })
                //
                //                RxJavaHooks.reset();
                //                RxJavaHooks.setOnIOScheduler(new Func1<Scheduler, Scheduler>() {
                //                    @Override
                //                    public Scheduler call(Scheduler scheduler) {
                //                        return Schedulers.immediate();
                //                    }
                //                });

                base.evaluate()
            }
        }
    }
}
